### PR TITLE
refactor: adding CORs headers to all responses using middy

### PR DESCRIPTION
**Pull request by andreabadesso**
# Motivation

The `wallet-desktop` requests to the wallet-service gets a [cross-origin request blocked error](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/CORS/Errors) if the server does not respond with a `Access-Control-Allow-Origin` header on the response

I'm also adding a very nice [middleware library](https://github.com/middyjs/middy) that we can use for other things in the wallet service, like parameter validation, MySQL connection sharing, authorization, and others later on

### Acceptance Criteria
- Our responses should include the `Access-Control-Allow-Origin` header and it should be set to `*`

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -30,6 +30,15 @@ plugins:
 
 resources:
   Resources:
+    GatewayResponseDefault4XX:
```
Comment (andreabadesso) [inline serverless.yml:36]: This is needed to be able to include the CORs headers on our custom authorizer

Taken from:
https://serverless.com/blog/cors-api-gateway-survival-guide/

File: package.json [hunk 2]
```diff
@@ -39,11 +41,12 @@
     "uuid": "^8.3.0"
   },
   "devDependencies": {
-    "@types/aws-lambda": "^8.10.76",
+    "@types/aws-lambda": "^8.10.95",
     "@types/jest": "^26.0.24",
     "@types/node": "^10.12.18",
     "@typescript-eslint/eslint-plugin": "^3.3.0",
     "@typescript-eslint/parser": "^3.3.0",
+    "bitcore-lib": "^8.25.25",
```
Comment (r4mmer) [inline package.json:49]: Why are we adding bitcore-lib as a dependency?

Comment (andreabadesso) [reply package.json:49]: We are not adding it in this PR, the diff is just a change of position in the object keys order

We have bitcore lib as a dev dependency to be able to sign messages in the same way the lib does, but we don't use it in production

File: tests/api.test.ts [hunk 3]
```diff
@@ -1161,3 +1210,38 @@ test('GET /wallet/tokens/token_id/details', async () => {
...
+  // test CORS headers
+  await _testCORSHeaders(txProposalSend, null, null);
+});
+
+test('DELETE /tx/proposal/{txProposalId}', async () => {
+  expect.hasAssertions();
+
+  // test CORS headers
+  await _testCORSHeaders(txProposalDestroy, null, null);
+});
```
Comment (andreabadesso) [inline tests/api.test.ts]: Those methods are being tested on different test files, so I 'm just adding tests for the CORs headers here to make sure they are all responding with the proper headers

File: serverless.yml (hunk 1 above)
Comment (luislhl) [reply serverless.yml:36]: Maybe we should add this as a comment in the code

**Review by r4mmer — APPROVED** [review]

**Review by luislhl — APPROVED** [review]

**Pull request merged**